pydoctor/templatewriter/pages/__init__.py

```diff
@@ -64,12 +64,35 @@ def format_decorators(obj: Union[model.Function, model.Attribute]) -> Iterator["
 
         yield '@', stan.children, tags.br()
 
-def format_signature(function: model.Function) -> "Flattenable":
+def format_signature(func: Union[model.Function, model.FunctionOverload]) -> "Flattenable":
     """
     Return a stan representation of a nicely-formatted source-like function signature for the given L{Function}.
     Arguments default values are linked to the appropriate objects when possible.
     """
-    return html2stan(str(function.signature)) if function.signature else "(...)"
+    return html2stan(str(func.signature)) if func.signature else "(...)"
+
+def format_overloads(func: model.Function) -> Iterator["Flattenable"]:
+    """
+    Format a function overloads definitions as nice HTML signatures.
+    """
+    for overload in func.overloads:
+        yield '@overload', tags.br(), tags.div(format_function_def(func.name, func.is_async, overload))
```

Comment:
We are loosing the initial decorator formatting here. It would be good to use `format_decorators()` with `FunctionOverload` objects too such that we can write the following:

```suggestion
        yield from format_decorators(overload)
        yield tags.div(format_function_def(func.name, func.is_async, overload))
```

This will also include links to the `overload` docs if one has configured the standard library intersphinx URL.

Reply:
The only problem here I see is that `format_decorators` expects its param to have the ability to `report()` warnings, so I will probably add a `report()` method on it. So I either have to add `report()` to `FunctionOverload` or do some other approach to reporting issues.

Reply:
Ah yeah… and report expect the system… looks like you’ll have to pass the system along… though we should not duplicate the reporting logic. Not easy

Reply:
Or maybe the `FunctionOverload` should keep a reference to the `Function` object and act like a proxy for the report() method.